rfcs/experimental/dynamic_selection/README.md

```diff
@@ -0,0 +1,516 @@
+
+#### select heuristic (expository)
+
+```cpp
+  template<typename ...Args>
+  selection_type fixed_resource_policy::select(Args&&...) {
+    if (initialized_) {
+      return selection_type{*this, resources_[fixed_offset_]};
+    } else {
+      throw std::logic_error(“select called before initialize”);
+    }
+  }
+```
+
+#### execution info reporting requirements
+
+none
+
+#### Exceptions
+
+Constructor or initialize may throw `std::bad_alloc` or `std::logic_error`.
+
+<a id="round_robin_id"></a>
+### `round_robin_policy`
+
+```cpp
+template<typename Backend=sycl_backend> 
+class round_robin_policy;
+```
+
+| Constructors and Initialization |
+| -----------------------------------------------------------|
+| `round_robin_policy(deferred_initialization_t); // (1)` |
+| `round_robin_policy(size_t offset=0); // (2)` |
+| `round_robin_policy(const std::vector<resource_t<Backend>>& resources); // (3)` |
+| `void initialize(size_t offset=0); // (4)` |
+| `void initialize(const std::vector<resource_t<Backend>>& resources); // (5)` |
+
+1. Defers initialization and requires a later call to `initialize`.
+2. Rotates through the default set of resources at each call to `select` beginning with `offset`.
+3. Uses the provided set of resources and rotates through the default set of resources at each call to `select`.
+4. Rotates through the default set of resources at each call to `select`.
+5. Uses the provided set of resources and rotates through the default set of resources at each call to `select`.
+
+`initialize` should only be called for policies that are constructed with `deferred_initialization_t`.
+
+#### select heuristic (expository)
+
+```cpp
+  template<typename ...Args>
+  selection_type select(Args&&...) {
+    if (initialized_) {
+      resources_size_t i = 0;
+      {
+        std::lock_guard<mutex_type> l(mutex_);
+        if (next_context_ == MAX_VALUE) {
+          next_context_ = MAX_VALUE%num_contexts_;
+        }
+        i = next_context_++ % num_contexts_;
+      }
+      auto &r = resources_[i];
+      return selection_type{*this, r};
+    } else {
+      throw std::logic_error(“select called before initialization”);
+    }
+  }
+```
+
+#### execution info reporting requirements
+
+none
+
+#### Exceptions
+
+Constructor or initialize may throw `std::bad_alloc` or `std::logic_error`.
+
+<a id="dynamic_load_id"></a>
+### `dynamic_load_policy`
+
+```cpp
+template<typename Backend=sycl_backend> 
+class round_robin_policy;
+```
+
+| Constructors and Initialization |
+| -----------------------------------------------------------|
+| `dynamic_load_policy(deferred_initialization_t); // (1)` |
+| `dynamic_load_policy(size_t offset=0); // (2)` |
+| `dynamic_load_policy(const std::vector<resource_t<Backend>>& resources); // (3)` |
+| `void initialize(size_t offset=0); // (4)` |
+| `void initialize(const std::vector<resource_t<Backend>>& resources); // (5)` |
```

Comment:
Again it seems no optional `offset` parameter is implemented.

Reply:
Ok, will fix.